---
title: "Lab meeting"
kernelspec:
  name: python3
  display_name: python3
jupytext:
  text_representation:
    extension: .md
    format_name: myst
    format_version: '0.13'
    jupytext_version: 1.13.8
---

2022-10-15

Tang Ziya
<!--
> 2022-10-15 | {sub-ref}`wordcount-words` words | {sub-ref}`wordcount-minutes` min read
-->

## What I did last week?

1. Further study in `Relate` and `MSMC2`
2. Process the new batch data

## What's next?

1. Determine the ALL `Relate` parameters' effect on the final result.
2. Write more test for my wrapper.

## Relate

The `.ped` and `.map` files were convert to `.vcf` via `plink2`.
<!--
```console
$ plink2 --pedmap snp.pure --rice --recode vcf --out relic
$ plink2 --pedmap ancestrycluster --rice --recode vcf --out anc
```
-->
```{code-cell}
:tags: [remove-input]
import pandas as pd
from myst_nb import glue
from matplotlib import pyplot as plt

vcf1 = pd.read_csv("/Users/tcztzy/Downloads/ancestrycluster/relic.vcf", skiprows=18, sep=r"\s+")
vcf2 = pd.read_csv("/Users/tcztzy/Downloads/clus_anc/clus_anc.vcf", skiprows=18, sep=r"\s+").query("REF != 'N'")
vcf3 = vcf1.merge(vcf2, on=["#CHROM", "POS", "ID"])

samples = ["PR04838", "PR05216", "PR05225", "PR05299", "PR05475"]
diff = pd.concat([(vcf3[f"{sample}_{sample}_x"] != vcf3[f"{sample}_{sample}_y"]) | ((vcf3[f"{sample}_{sample}_x"] == ".") & (vcf3[f"{sample}_{sample}_y"] == ".")) for sample in samples], axis=1)
n_diff = (diff.all(axis=1)).sum()
glue("vcf1_len", len(vcf1), display=False)
glue("vcf2_len", len(vcf2), display=False)
glue("vcf3_len", len(vcf3), display=False)
glue("n_diff", n_diff, display=False)
```
<!--
The number of SNPs in first batch is {glue}`vcf1_len`, and {glue}`vcf2_len` in second batch, there are {glue}`vcf3_len` SNPs in both.

Second batch doesn't include all SNPs in first batch may cause by different filter applied.

There are {glue}`n_diff` SNPs invert the major and minor alleles because of the population frequency differ in two batches.
```{revealjs-break}
:notitle:
```
```{code-cell}
:tags: [remove-input]
from matplotlib import pyplot as plt
from matplotlib_venn import venn3
venn3(
    subsets=(len(vcf1) - len(vcf3), len(vcf2) - len(vcf3), len(vcf3) - n_diff, 0, 0, 0, n_diff),
    set_labels=("Batch 1", "Batch 2", "invert")
)
plt.show()
```
-->



## Parameters

### Input files `.haps`/`.sample`

Lack of INDEL support. There are {glue}`n_missing` of {glue}`vcf2_len` SNPs contain at least one missing allele.

```{code-cell}
:tags: [remove-input]
import missingno as msno
cols = [
    "PR04838",
    "PR04849",
    "PR04853",
    "PR05216",
    "PR05225",
    "PR05227",
    "PR05242",
    "PR05287",
    "PR05299",
    "PR05300",
    "PR05359",
    "PR05407",
    "PR05408",
    "PR05475",
    "PR05519",
]
missing = vcf2.query(" or ".join(f"'.' in {col}_{col}" for col in cols))
glue("n_missing", len(missing), display=False)
msno.matrix(vcf2.replace({".": None}).rename(columns={f"{col}_{col}": col for col in cols})[cols])
plt.show()
```

<!--
| Parameter | Description | Effects |
|-----------|-------------|---------|
| `--haps`, `--sample` | input files |  |
| `--map` | Genetic map | High quality genetic map will help |
| `--mutation_rate` | Mutation rate | I choose 1.2e-8 from literature. |
| `--years_per_gen` | Years per generation | I think 1 year per generation is reasonable. |
| `--effectiveN`/`--coal` |  |  |
-->
